Extension/src/LanguageServer/configurations.ts

```diff
@@ -736,34 +737,57 @@ export class CppProperties {
     }
 
     private resolveAndSplit(paths: string[] | undefined, defaultValue: string[] | undefined, env: Environment, glob = false): string[] {
-        const result: string[] = [];
-        if (paths) {
-            paths = this.resolveDefaults(paths, defaultValue);
-            paths.forEach(entry => {
-                const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
-                result.push(...entries);
-            });
-        }
+        const resolvedVariables: string[] = [];
+        if (paths === undefined) {
+            return resolvedVariables;
+        }
+        paths = this.resolveDefaults(paths, defaultValue);
+        paths.forEach(entry => {
+            const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
+            resolvedVariables.push(...entries);
+        });
         if (!glob) {
-            return result;
-        }
-
-        const globResult: string[] = [];
-        for (let res of result) {
-            // fastGlob will expand the ending double wildcard. temporary strip them before expanding
-            const recursive: boolean = res.endsWith('**');
-            if (recursive) {
-                res = res.slice(0, res.length - 2);
+            return resolvedVariables;
+        }
+        const resolvedGlob: string[] = [];
+        for (let res of resolvedVariables) {
+            let counter: number = 0;
+            let slashFound: boolean = false;
+            const lastIndex: number = res.length - 1;
+            // Detect and glob all wildcard variations by looking at last character in the path first.
+            for (let i: number = lastIndex; i >= 0; i--) {
+                if (res[i] === '*') {
+                    counter++;
```

Comment:
Is it valid to have more than 2 "*'?  What should happen if more than 2 are present?

Reply:
Seems fine to me?

Reply:
I am assuming that would result in an invalid path, the code would parse it but glob would return an empty result.

Reply:
But this case is only for *** at the end, which isn't passed to fastGlob, which is fine.